src/Classes/Particle/Linker.ts

```diff
@@ -0,0 +1,102 @@
+
+                    /* particles.line_linked - convert hex colors to rgb */
+                    //  check for the color profile requested and
+                    //  then return appropriate value
+
+                    if (color === "random") {
+                        if (options.particles.line_linked.consent) {
+                            container.particles.lineLinkedColor = Utils.hexToRgb(color);
+                        } else if (options.particles.line_linked.blink) {
+                            container.particles.lineLinkedColor = "random";
+                        } else {
+                            container.particles.lineLinkedColor = "mid";
+                        }
+                    } else {
+                        container.particles.lineLinkedColor = Utils.hexToRgb(color);
+                    }
+                }
+
+                if (!container.canvas.context) {
+                    return;
+                }
+
+                const ctx = container.canvas.context;
+
+                let colorLine: IRgb | undefined;
+
+                /*
+                 * particles connecting line color:
+                 *
+                 *  random: in blink mode : in every frame refresh the color would change
+                 *          hence resulting blinking of lines
+                 *  mid: in consent mode: sample particles color and get a mid level color
+                 *                        from those two for the connecting line color
+                 */
+
+                if (container.particles.lineLinkedColor === "random") {
+                    colorLine = Utils.getRandomColorRGBA();
+                } else if (container.particles.lineLinkedColor == "mid" && particle.color && p2.color) {
+                    const sourceColor = particle.color;
+                    const destColor = p2.color;
+
+                    colorLine = {
+                        b: Math.floor(Utils.mixComponents(sourceColor.b, destColor.b, particle.radius, p2.radius)),
+                        g: Math.floor(Utils.mixComponents(sourceColor.g, destColor.g, particle.radius, p2.radius)),
+                        r: Math.floor(Utils.mixComponents(sourceColor.r, destColor.r, particle.radius, p2.radius)),
+                    };
+                } else {
+                    colorLine = container.particles.lineLinkedColor as IRgb;
+                }
+
+                if (colorLine) {
+                    ctx.strokeStyle = `rgba(${colorLine.r},${colorLine.g},${colorLine.b},${opacityLine})`;
+                }
+
+                ctx.lineWidth = options.particles.line_linked.width;
+                // container.canvas.ctx.lineCap = "round"; /* performance issue */
+                /* path */
+                ctx.beginPath();
+                ctx.moveTo(x1, y1);
+                ctx.lineTo(x2, y2);
+                ctx.stroke();
+                ctx.closePath();
+            }
+        }
+    }
+}
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [file should end with a newline](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5110920)